From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Wed, 30 Jul 2014 15:20:19 +0200
Subject: [PATCH] gpu: ipu-v3: Register scaler platform device

This patch registers the scaler device using the IC post-processing task,
to be handled by a mem2mem scaler driver.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/gpu/ipu-v3/ipu-common.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/ipu-v3/ipu-common.c b/drivers/gpu/ipu-v3/ipu-common.c
index 03ea3e7a2967..f39e2f6f704e 100644
--- a/drivers/gpu/ipu-v3/ipu-common.c
+++ b/drivers/gpu/ipu-v3/ipu-common.c
@@ -1037,6 +1037,8 @@ static struct ipu_platform_reg client_reg[] = {
 			.dma[1] = -EINVAL,
 		},
 		.name = "imx-ipuv3-crtc",
+	}, {
+		.name = "imx-ipuv3-scaler",
 	},
 };
 
@@ -1061,7 +1063,7 @@ static int ipu_add_client_devices(struct ipu_soc *ipu, unsigned long ipu_base)
 
 		/* Associate subdevice with the corresponding port node */
 		of_node = of_graph_get_port_by_id(dev->of_node, i);
-		if (!of_node) {
+		if (i < 4 && !of_node) {
 			dev_info(dev,
 				 "no port@%d node in %s, not using %s%d\n",
 				 i, dev->of_node->full_name,
